group/assignment-5/test/alphabet/graph/AlphabetDAGTest.java

```diff
@@ -0,0 +1,208 @@
+package alphabet.graph;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import alphabet.InvalidLexiconException;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import alphabet.graph.AlphabetDAG.CharNode;
+
+/** Using simple tests to verify correct implementation of AlphabetDAG. */
+@RunWith(JUnit4.class)
+public final class AlphabetDAGTest {
+  private static final int REPETITIONS = 2_000_000;
+
+  private AlphabetDAG alphabetDAGInvalid;
+  private AlphabetDAG alphabetDAG;
+  private List<String> dictionary;
+
+  @Test(expected = IllegalArgumentException.class)
+  public void invalidConstructionNullDictionary() {
+    alphabetDAGInvalid = AlphabetDAG.buildGraphFromDictionary(null);
+  }
+
+  @Test(expected = IllegalArgumentException.class)
+  public void invalidConstructionNullWord() {
+    alphabetDAGInvalid = AlphabetDAG.buildGraphFromDictionary(Arrays.asList("a", "b", null));
+  }
+
+  @Test(expected = IllegalArgumentException.class)
+  public void invalidConstructionEmptyDic() {
+    alphabetDAGInvalid = AlphabetDAG.buildGraphFromDictionary(new ArrayList<String>());
```

Comment:
```suggestion
    alphabetDAGInvalid = AlphabetDAG.buildGraphFromDictionary(new ArrayList<>());
```

I think we're using java 8 so, we don't have add explicit type here

Reply:
Oh right, I wasn't aware It can be done without using the explicit type in a declared variable first.